save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/tables/TableComponent.kt

```diff
@@ -126,7 +126,15 @@ fun <D : Any, P : TableProps<D>> tableComponent(
         additionalOptions()
     }, plugins = plugins)
 
-    useEffect(tableInstance.state.pageSize) {
+    // list of entities, updates of which will cause update of the data retrieving effect
+    val dependencies: Array<dynamic> = if (useServerPaging) {
+        arrayOf(tableInstance.state.pageIndex, tableInstance.state.pageSize, pageCount)
+    } else {
+        // when all data is already available, we don't need to repeat `getData` calls
+        emptyArray()
+    } + getAdditionalDependencies(props)
+
+    useEffect(*dependencies) {
```

Comment:
Did you figure out exactly which values from dependencies were required? `pageCount`? Or others too?

Reply:
All of them 😐